global.go

```diff
@@ -128,6 +127,46 @@ func RedirectStdLog(l *Logger) func() {
 	}
 }
 
+// RedirectStdLogAt redirects output from the standard library's package-global
+// logger to the supplied logger at required level. Since zap already handles
+// caller annotations, timestamps, etc., it automatically disables the standard
+// library's annotations and prefixing.
+//
+// It returns a function to restore the original prefix and flags and reset the
+// standard library's output to os.Stdout.
+func RedirectStdLogAt(l *Logger, level zapcore.Level) (func(), error) {
+	flags := log.Flags()
+	prefix := log.Prefix()
+	log.SetFlags(0)
+	log.SetPrefix("")
+	logger := l.WithOptions(AddCallerSkip(_stdLogDefaultDepth + _loggerWriterDepth))
+	var logFunc func(string, ...zapcore.Field)
+	switch level {
+	case DebugLevel:
```

Comment:
I can imagine a `levelToFunc` function here to DRY.

Reply:
Agreed! You've been patient enough, though - I'm happy to merge without that and fix it myself :)